src/Text/Microstache/Type.hs

```diff
@@ -126,3 +128,20 @@ instance Exception MustacheException where
 #else
 instance Exception MustacheException
 #endif
+
+data MustacheWarning
+  = MustacheVariableNotFound Key
+    -- ^ The template contained a variable for which there was no data counterpart in the current context
+  | MustacheDirectlyRenderedValue Key
+    -- ^ A complex value such as an Object or Array was directly rendered into the template
+  deriving (Eq, Show, Typeable, Generic)
+
+#if MIN_VERSION_base(4,8,0)
+instance Exception MustacheWarning where
+  displayException (MustacheVariableNotFound key) = 
+    "Referenced value was not provided, key: " ++ T.unpack (showKey key)
+  displayException (MustacheDirectlyRenderedValue key) = 
+    "Complex value rendered as such, key: " ++ T.unpack (showKey key)
```

Comment:
Can you factor this `displayException` implementation out to a top-level function? It'd be handy to be able to use this on pre-`base-4.8` compilers.